x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/ip/IpPrefixTests.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function.scalar.ip;
+
+import com.carrotsearch.randomizedtesting.annotations.Name;
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+
+import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataTypes;
+import org.elasticsearch.xpack.esql.expression.function.AbstractFunctionTestCase;
+import org.elasticsearch.xpack.esql.expression.function.FunctionName;
+import org.elasticsearch.xpack.esql.expression.function.TestCaseSupplier;
+import org.elasticsearch.xpack.esql.type.EsqlDataTypeConverter;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+import static org.hamcrest.Matchers.equalTo;
+
+@FunctionName("ip_prefix")
+public class IpPrefixTests extends AbstractFunctionTestCase {
+    public IpPrefixTests(@Name("TestCase") Supplier<TestCaseSupplier.TestCase> testCaseSupplier) {
+        this.testCase = testCaseSupplier.get();
+    }
+
+    @ParametersFactory
+    public static Iterable<Object[]> parameters() {
+
+        var suppliers = List.of(
+            new TestCaseSupplier(
+                List.of(DataTypes.IP, DataTypes.INTEGER),
+                () -> new TestCaseSupplier.TestCase(
+                    List.of(
+                        new TestCaseSupplier.TypedData(EsqlDataTypeConverter.stringToIP("1.2.3.4"), DataTypes.IP, "ip"),
+                        new TestCaseSupplier.TypedData(24, DataTypes.INTEGER, "prefixLength")
+                    ),
+                    "IpPrefixEvaluator[ip=Attribute[channel=0], prefixLength=Attribute[channel=1]]",
+                    DataTypes.IP,
+                    equalTo(EsqlDataTypeConverter.stringToIP("1.2.3.0"))
+                )
+            ),
+            new TestCaseSupplier(
+                List.of(DataTypes.IP, DataTypes.INTEGER),
+                () -> new TestCaseSupplier.TestCase(
+                    List.of(
+                        new TestCaseSupplier.TypedData(EsqlDataTypeConverter.stringToIP("::ff"), DataTypes.IP, "ip"),
+                        new TestCaseSupplier.TypedData(127, DataTypes.INTEGER, "prefixLength")
+                    ),
+                    "IpPrefixEvaluator[ip=Attribute[channel=0], prefixLength=Attribute[channel=1]]",
+                    DataTypes.IP,
+                    equalTo(EsqlDataTypeConverter.stringToIP("::fe"))
+                )
+            )
+        );
+
+        return parameterSuppliersFromTypedData(anyNullIsNull(true, suppliers));
```

Comment:
Could you add `errorsForCasesWithoutExamples` too? That just makes sure that if we didn't mention some sequence of parameters they result in a sane-ish error message. It also bumps the number of tests up quite a bit which is sad, but it is useful.

Reply:
I actually removed it explicitly, as seen in other tests.

The problem is, it tries to fail by using a long in an integer param. But it actually works. Checking some other functions, I saw they were working this way. But they could be outdated.

Should we add all integer types here? That is, short, integer, long and unsigned long. I've mixed feelings here around if the engine should downcast automatically and we should forget about all bigger types. Making functions is quite a chore already, and feels weird to add more things.

This is just a side effect of using `errorsForCasesWithoutExamples`, so I need more info on the topic/decisions here

Reply:
Oh neat, it found a bug! Or, at least something to talk about. If you don't want to support `long` here then you have to make a change to the function implementation not to. The `isInteger` calls check for any integer-like-type. There's some other way to force just an integer.